### ISSUE TITLE
Enhancement: Explicit selection of the datatype for query parameters

### ISSUE THREAD

**Issue by rafek1241** [mikependon/RepoDb#1052]
### Describe the enhancement

I would like to have better control over query parameters passed to execute against RAW SQL:

```csharp
var parameter = new Parameter("@Id", "10045", DBType.AnsiString, false);
var fields = new[]{ new QueryField(parameter) };
var queryGroup = new QueryGroup(fields);

var result = connection.ExecuteQuery("<SOME SQL> WHERE Id = @Id", queryGroup);

/* Produces SQL:
EXEC sp_executesql '...' @id VARCHAR
*/

```
OR
```csharp
var result = connection.ExecuteQuery("<SOME SQL> WHERE Id = @Id", new { Id = new Parameter("<value>",  DBType.AnsiString) });

/* Produces SQL:
EXEC sp_executesql '...' @id VARCHAR
*/
```
OR by using implementation from ADO.NET
```csharp
var result = connection.ExecuteQuery("<SOME SQL> WHERE Id = @Id", new []{ new  Microsoft.Data.SqlClient.SqlParameter("Id", SqlDbType.VarChar) });

/* Produces SQL:
EXEC sp_executesql '...' @id VARCHAR
*/
```

Reference: https://repodb.net/class/parameter

Reference to the dapper's solution: https://www.learndapper.com/parameters
```csharp
var parameters = new DynamicParameters();
var customerId = "ALFKI";
parameters.Add("@CustomerId", customerId, DbType.String, ParameterDirection.Input, customerId.Length);
var sql = "select * from customers where CustomerId = @CustomerId";
using (var connection = new SqlConnection(connString))
{
    var customer = connection.QuerySingle<Customer>(sql, parameters);
}
```

### Context

In the company that I'm working I can use only raw SQLs and I would like to have better control of the passed parameters to the query statement to be able to select datatype. As @mikependon mentioned in previous issue that I made (#1051), RepoDB is probably leveraging the default behavior of ADO.NET for RAW SQL. Because the tables in the database uses `VARCHAR` instead of `NVARCHAR`, I've got a problem with slow performance of the queries. Things that should take 5-7 seconds, take 40-50s just because of different datatype..
I mitigate the problem by using:
```csharp            
var @params = new
            {
                Id = id
            };

            TypeMapper.Add<string>(DbType.AnsiString);
            var data = await connection.ExecuteQueryMultipleAsync(query, @params);
            TypeMapper.Remove<string>();
```
BUT when there will be two parameters of same CLR .NET type with different datatypes in sql - I will be ..... up.

**Comment by mikependon** [comment]
Referencing #1060

**Comment by mikependon** [comment]
The fixes to this will be available on the next version (> RepoDb v1.10.12).
